CMakeLists.txt

```diff
@@ -884,9 +886,11 @@ if(NOT WIN32 OR ROCKSDB_INSTALL_ON_WINDOWS)
   )
 endif()
 
+add_subdirectory(third-party/gtest-1.8.1/fused-src/gtest)
 option(WITH_TESTS "build with tests" ON)
-if(WITH_TESTS)
-  add_subdirectory(third-party/gtest-1.8.1/fused-src/gtest)
+# For test libraries, utilities, and exes that are build iff WITH_TESTS=ON and
+# in Debug mode. Add test only code that is not #ifdefed for Release here.
+if(WITH_TESTS AND CMAKE_BUILD_TYPE STREQUAL "Debug")
```

Comment:
@siying @riversand963 Finally passed the ci checks, perhaps you may confuse about `CMakeLists.txt` changes. To be brief, I'm not convinced of including benchmark part in test part. As a rocksdb user, I'm trying to build in release mode along with `db_bench`, but I **HAVE TO** work with the option `WITH_TESTS=ON`, while comments like _For test util library that is build only in DEBUG mode_ (maybe `CMAKE_BUILD_TYPE=Debug`)diverge with it at the same time. So I prefer to isolate the benchmark part from the test part with a new option `WITH_BENCHMARK_TOOLS`, and make use of release/debug mode with respect to the following comments. Could you plz have a review?